src/node-red/getting-started/programming/loop.md

```diff
@@ -0,0 +1,205 @@
+---
+eleventyNavigation:
+  key: Loop
+  parent: Programming
+meta:
+  title: How to implement loops in Node-RED flows
+  description: Learn how to implement while, for, and for...of loops in Node-RED with core and custom nodes for efficient data processing and automation.
+  keywords: node red loop, nodered loop, node red for loop, node-red loop, nodered for of, node-red while loop, node-red do while loop
+---
+
+# {{meta.title}}
+
+Handling repetitive tasks is a common challenge in automation and data processing. Whether you need to iterate over large datasets, perform calculations, or execute operations based on conditions multiple times, using loops can significantly enhance efficiency and scalability.
+
+In this document, we’ll explore how to replicate different types of loops that are essential in various contexts. We’ll discuss their applications and provide examples to help you effectively implement them in Node-RED.
+
+## What is a Loop?
+
+A [loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Loops_and_iteration) is a programming construct that allows you to execute a block of code repeatedly until a certain condition is met. Different types of loops are suited to different scenarios, depending on how and when you want the code to repeat:
+
+- **For Loop**: Executes a block of code a specific number of times. This is useful when you know in advance how many times you need to iterate, such as iterating through a range of numbers or items in a list. It's also known as a fixed number loop.
+
+- **While Loop**: Repeats a block of code as long as a specified condition is true. This type of loop is useful when you don’t know how many times the loop will need to run beforehand. The loop continues executing until the condition becomes false.
+
+- **Do...While Loop**: Similar to the while loop, but with a key difference: it guarantees that the code block will execute at least once before checking the condition. This is because the condition is evaluated after the loop’s code has executed.
+
```

Comment:
Do while and a while loop are the same. Feels like we should differentiate the 3 cases:
1. For Each - Iterate over a set of elements and perform a task on each of them
2. While - Perform a task until a certain exit criteria is met
3. For - Execute a certain block a certain numer of times.